crates/fuel-core/src/graphql_api/storage/coins.rs

```diff
@@ -2,16 +2,34 @@ use fuel_core_storage::{
     blueprint::plain::Plain,
     codec::{
         postcard::Postcard,
-        primitive::utxo_id_to_bytes,
+        primitive::{
+            utxo_id_to_bytes,
+            Primitive,
+        },
         raw::Raw,
     },
     structured_storage::TableWithBlueprint,
     Mappable,
 };
-use fuel_core_types::fuel_tx::{
-    Address,
-    TxId,
-    UtxoId,
+use fuel_core_types::{
+    entities::{
+        coins::coin::Coin,
+        Message,
+    },
+    fuel_tx::{
+        Address,
+        AssetId,
+        TxId,
+        UtxoId,
+    },
+    fuel_types::Nonce,
+};
+
+use crate::graphql_api::indexation;
+
+use self::indexation::coins_to_spend::{
+    NON_RETRYABLE_BYTE,
```

Comment:
In another comment I mentioned, that we don't need to include retryable messages into the coins to spend query. So we can remove it.

But you need to know the difference between coin and message, so I think we need to use this 1 byte for the `Message` or `Coin` enum representation.